### DISCUSSION TITLE
$$Props error in Svelte

### DISCUSSION THREAD

**Discussion by ghost** [xyflow/xyflow#4636] [Q&A] [answered]
so, i just tried out svelte-flow and following CustomNode example, but i get this error

![Screenshot 2024-09-10 103614](https://github.com/user-attachments/assets/c8799258-f82f-4436-a3d6-9aba9e4d1ddd)

how to fix this?

note: the app still running, i just wanna get rid of this error

**Comment by moklick** [comment]
We might need to update our docs. Could you try:

```js
type $$Props = NodeProps<{ color: Writable<string> }>; ?
```

**Reply by ghost** [reply]
still error after applying

```
Type '{ color: Writable<string>; }' does not satisfy the constraint 'Node<Record<string, unknown>, string>'.
  Type '{ color: Writable<string>; }' is missing the following properties from type 'NodeBase<Record<string, unknown>, string>': id, position, datats(2344)
```

oh, my code just look like this
```typescript
import { Handle, Position, type NodeProps } from "@xyflow/svelte";
import type { Writable } from "svelte/store";

type $$Props = NodeProps<{ color: Writable<string> }>;

export let data: { color: Writable<string> };

const { color } = data;
```

**Reply by bcakmakoglu** [reply, marked as answer]
I believe it's supposed to be typed like this: `type $$Props = NodeProps<Node<{ color: Writable<string> }>>;`
